app/__mocks__/recoil.ts

```diff
@@ -0,0 +1,127 @@
+import * as recoil from "recoil";
+export * from "recoil";
+
+export let mockValues = {};
+export let mockValuesStore = {};
+export let mockDefaults = {};
+export function setMockAtoms(newMockValues: { [key: string]: any }) {
+  mockValues = {
+    ...mockValues,
+    ...newMockValues,
+  };
+  mockDefaults = { ...mockDefaults, ...newMockValues };
+}
+
+const getValue = (atom) => {
+  if (mockValuesStore[atom.key]) {
+    const str = JSON.stringify(atom.params);
+    if (mockValuesStore[atom.key].hasOwnProperty(str)) {
+      return mockValuesStore[atom.key][JSON.stringify[str]];
+    }
+  }
+
+  if (atom.params !== undefined) {
+    console.log(atom.key);
+    return mockValues[atom.key](atom.params);
+  }
+  return mockValues[atom.key];
+};
+
+const resetValue = (atom) => {
+  if (atom.params && mockValuesStore[atom.key]) {
+    delete mockValuesStore[atom.key][JSON.stringify(atom.params)];
+  } else {
+    delete mockValues[atom.key];
+  }
+};
+
+const setValue = (atom, value) => {
+  if (atom.params) {
+    if (!mockValuesStore[atom.key]) mockValuesStore[atom.key] = {};
+    mockValuesStore[atom.key][JSON.stringify(atom.params)] = value;
+  } else {
+    mockValues[atom.key] = value;
+  }
+};
+
+export function atom<T>(options: Parameters<typeof recoil.atom<T>>[0]) {
+  return { key: options.key };
+}
+
+export function atomFamily<T, P extends recoil.SerializableParam>(
+  options: Parameters<typeof recoil.atomFamily<T, P>>[0]
+) {
+  return (params: P) => ({
+    key: options.key,
+    params: params,
+  });
+}
+
+export function selector<T extends unknown>(
+  options: Parameters<typeof recoil.selector<T>>[0]
+): { (): T; key: string } {
+  function resolver() {
+    return options.get({
+      get: getValue,
+      getCallback: (callback) => {
```

Comment:
IIUC this will allow for testing selectors as they will resolve through this mock impl of recoil. Is there any way to know for sure that this implementation behaves like the real recoil?

Reply:
The implementation is not correct, but I don't think it would be hard to implement. I've left it as a todo. Only `Snapshot` needs to be mocked to return mock values.